index.js

```diff
@@ -65,47 +66,59 @@ module.exports = function f (b, opts) {
                 outputs = [];
             }
 
+            // utility for creating output streams
             function moreOutputs (file) {
                 if (isarray(outopt)) return [];
                 if (!outopt) return [];
                 var xopts = { env: xtend(process.env, { FILE: file }) };
                 return [ outpipe(outopt, xopts) ];
             }
 
-            var pipelines = files.reduce(function (acc, x, ix) {
+            // create a pipeline (labeled-stream-splicer) for every entry file
+            // with only a packer
+            var pipelines = files.reduce(function (acc, file, index) {
                 var pipeline = splicer.obj([
                     'pack', [ pack(packOpts) ],
                     'wrap', []
                 ]);
 
-                if (ix >= outputs.length) {
-                    outputs.push.apply(outputs, moreOutputs(x));
+                // if not enough specified outputs, possibly add additional outputs (?)
+                if (index >= outputs.length) {
+                    outputs.push.apply(outputs, moreOutputs(file));
                 }
-                if (outputs[ix]) pipeline.pipe(outputs[ix]);
-                
-                acc[path.resolve(cwd, x)] = pipeline;
+                // pipe to output if one exists
+                if (outputs[index]) pipeline.pipe(outputs[index]);
+
+                acc[path.resolve(cwd, file)] = pipeline;
                 return acc;
             }, {});
 
             // Force browser-pack to wrap the common bundle
             b._bpack.hasExports = true;
 
+            // for tests/debugging
             Object.keys(pipelines).forEach(function (id) {
                 b.emit('factor.pipeline', id, pipelines[id]);
             });
 
+            // create factor stream
             var s = createStream(files, opts);
+
+            // connect each factor groups module stream to bundle pipeline
             s.on('stream', function (bundle) {
+                // each group's output stream
                 bundle.pipe(pipelines[bundle.file]);
             });
 
+            // add factor stream before pack
             b.pipeline.get('pack').unshift(s);
 
             if (needRecords) files = [];
 
             next();
         }));
 
+        // capture module's relative path. used for recovering entry point pathnames
```

Comment:
I would keep this comment but probably remove most of the others…tbh, I feel like many of those only restate the line and don't actually clarify much.

Reply:
I added the comments to stay sane while trying to read through the code which i found unintuitive